File: experiment_jp_test/j2pyraw/for__2__COUNT_ARRAYS_ADJACENT_ELEMENTS_ONE_DIVIDE_ANOTHER.java
class X { static int numofArray ( int n , int m ) { int [ ] [ ] dp = new int [ MAX ] [ MAX ] ; Vector < Integer > [ ] di = new Vector [ MAX ] ; Vector < Integer > [ ] mu = new Vector [ MAX ] ; for ( int i = 1 ; i <= m ; i ++ ) dp [ 1 ] [ i ] = 1 ; for ( int i = 2 ; i <= n ; i ++ ) { for ( int j = 1 ; j <= m ; j ++ ) { dp [ i ] [ j ] = 0 ; for ( Integer x : di [ j ] ) dp [ i ] [ j ] += dp [ i - 1 ] [ x ] ; for ( Integer x : mu [ j ] ) dp [ i ] [ j ] += dp [ i - 1 ] [ x ] ; } } int ans = 0 ; for ( int i = 1 ; i <= m ; i ++ ) { ans += dp [ n ] [ i ] ; di [ i ] . clear ( ) ; mu [ i ] . clear ( ) ; } return ans ; }
 }

File: experiment/java-python/test/j2pyrawpy/for__2__COUNT_ARRAYS_ADJACENT_ELEMENTS_ONE_DIVIDE_ANOTHER.py
#!/usr/bin/env python
""" generated source for module for__2__COUNT_ARRAYS_ADJACENT_ELEMENTS_ONE_DIVIDE_ANOTHER """
class X(object):
    """ generated source for class X """
    @classmethod
    def numofArray(cls, n, m):
        """ generated source for method numofArray """
        dp = [None]*MAX
        di = [None]*MAX
        mu = [None]*MAX
        i = 1
        while i <= m:
            i += 1
        i = 2
        while i <= n:
            while j <= m:
                dp[i][j] = 0
                for x in di[j]:
                    dp[i][j] += dp[i - 1][x]
                for x in mu[j]:
                    dp[i][j] += dp[i - 1][x]
                j += 1
            i += 1
        ans = 0
        i = 1
        while i <= m:
            ans += dp[n][i]
            di[i].clear()
            mu[i].clear()
            i += 1
        return ans

